scipy/spatial/distance.py

```diff
@@ -1837,7 +1835,7 @@ def _cosine_cdist(XA, XB, dm):
     dm += 1
 
 
-def cdist(XA, XB, metric='euclidean', p=2, V=None, VI=None, w=None):
+def cdist(XA, XB, metric='euclidean', p=None, w=None, V=None, VI=None):
```

Comment:
Dura lex sed lex. I reverted the change.

Anyway, I think that we might have made an exception in this particular case since the calls that would be broken are quite unlikely. In fact, they would all require dummy args:

    cdist(XA, XB, metric='minkowski', p)  # still works 
    cdist(XA, XB, metric='wminkowski', p, dummyarg, dummyarg, w)  # 2 dummy args
    cdist(XA, XB, metric='mahalanobis', dummyarg, dummyarg, VI)  # 2 dummy args
    cdist(XA, XB, metric='seuclidean', dummyarg, V)  # 1 dummy arg

p.s. this IMHO is another point for using **kwargs